sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jdbc/JdbcConnectionString.java

```diff
@@ -0,0 +1,136 @@
+
+        if (pathQueryDelimiterIndex < 0) {
+            return;
+        }
+
+        String hostInfo = this.jdbcURL.substring(databaseType.getSchema().length() + 3, pathQueryDelimiterIndex);
+        String[] hostInfoArray = hostInfo.split(":");
+        if (hostInfoArray.length == 2) {
+            this.properties.put("servername", hostInfoArray[0]);
+            this.properties.put("port", hostInfoArray[1]);
+        } else {
+            this.properties.put("port", hostInfo);
+        }
+
+        String properties = this.jdbcURL.substring(pathQueryDelimiterIndex + 1);
+
+        final String[] tokenValuePairs = properties.split(this.databaseType.getQueryDelimiter());
+
+        for (String tokenValuePair : tokenValuePairs) {
+            final String[] pair = tokenValuePair.split(TOKEN_VALUE_SEPARATOR, 2);
+            String key = pair[0];
+            if (!StringUtils.hasText(pair[0])) {
+                throw new IllegalArgumentException(String.format(INVALID_PROPERTY_PAIR_FORMAT, tokenValuePair));
+            }
+            if (pair.length < 2) {
+                this.properties.put(key, NONE_VALUE);
+            } else {
+                this.properties.put(key, pair[1]);
+            }
+        }
+    }
+
+    public String enhanceConnectionString(Map<String, String> enhancedProperties) {
+        if (enhancedProperties == null || enhancedProperties.isEmpty()) {
+            return this.jdbcURL;
+        }
+        LOGGER.debug("Trying to enhance url for {}", databaseType);
+
+        StringBuilder builder = new StringBuilder(this.jdbcURL);
+
+        if (!this.hasProperties()) {
+            builder.append(databaseType.getPathQueryDelimiter());
+        } else {
+            builder.append(databaseType.getQueryDelimiter());
+        }
+
+        for (Map.Entry<String, String> entry : enhancedProperties.entrySet()) {
+            String key = entry.getKey(), value = entry.getValue();
+            String valueProvidedInConnectionString = this.getProperty(key);
+
+            if (valueProvidedInConnectionString == null) {
+                builder.append(key)
+                    .append("=")
+                    .append(value)
+                    .append(databaseType.getQueryDelimiter());
+            } else if (!value.equals(valueProvidedInConnectionString)) {
+                LOGGER.debug("The property {} is set to another value than default {}", key, value);
+                throw new IllegalArgumentException("Inconsistent property detected");
+            } else {
+                LOGGER.debug("The property {} is already set", key);
+            }
+        }
+
+        String enhancedUrl = builder.toString();
+        return enhancedUrl.substring(0, enhancedUrl.length() - 1);
+    }
+
+    public String getProperty(String key) {
+        return this.properties.get(key);
+    }
+
+    public DatabaseType getDatabaseType() {
+        return databaseType;
+    }
+
+    public boolean hasProperties() {
```

Comment:
what does this method mean?

if a `JdbcConnectionString` has properties? what is a *property* from a `JdbcConnectionString` perspective?